File: ws/quiroga/u2/Ws15Inheritance/ZooApp/src/main/java/ec/edu/espe/zooapp/model/Animal.java

package ec.edu.espe.zooapp.model;

import java.util.ArrayList;
import java.util.Date;

/**
 *
 * @author Esteban Quiroga 
 */
public class Animal {
    protected int id;
    private String name;
    private Date bornOnDate;
    private float weigth;
    private char sex;
    private ArrayList<Vaccine> vaccines;
    
    public void feed(Food food){
        System.out.println("Feeding the animal with --> " + food);
    }
    
    public void register(){
        System.out.println("Registering animal...");
    }
    
    public void assignCage(Cage cage, Animal animal){
        //TODO assign animal to a cage
        System.out.println("Assigning an animal " + animal+ " to cage --> " + cage);
        cage.getAnimals().add(this);
    }
     
    public void provideVaccination(Vaccine vaccine){
        System.out.println("Providing vaccine --> " + vaccine + " <-- to "+ name);
        vaccines.add(vaccine);
    }

    public Animal(int id, String name, Date bornOnDate, float weigth, char sex, ArrayList<Vaccine> vaccines) {
        this.id = id;
        this.name = name;
        this.bornOnDate = bornOnDate;
        this.weigth = weigth;
        this.sex = sex;
        this.vaccines = vaccines;
    }

    @Override
    public String toString() {
        return "Animal{" + "id=" + id + ", name=" + name + ", bornOnDate=" + bornOnDate + ", weigth=" + weigth + ", sex=" + sex + ", vaccines=" + vaccines + '}';
    }

    
    /**
     * @return the id
     */
    public int getId() {
        return id;
    }

    /**
     * @param id the id to set
     */
    public void setId(int id) {
        this.id = id;
    }

    /**
     * @return the name
     */
    public String getName() {
        return name;
    }

    /**
     * @param name the name to set
     */
    public void setName(String name) {
        this.name = name;
    }

    /**
     * @return the bornOnDate
     */
    public Date getBornOnDate() {
        return bornOnDate;
    }

    /**
     * @param bornOnDate the bornOnDate to set
     */
    public void setBornOnDate(Date bornOnDate) {
        this.bornOnDate = bornOnDate;
    }

    /**
     * @return the weigth
     */
    public float getWeigth() {
        return weigth;
    }

    /**
     * @param weigth the weigth to set
     */
    public void setWeigth(float weigth) {
        this.weigth = weigth;
    }

    /**
     * @return the sex
     */
    public char getSex() {
        return sex;
    }

    /**
     * @param sex the sex to set
     */
    public void setSex(char sex) {
        this.sex = sex;
    }

    /**
     * @return the vaccines
     */
    public ArrayList<Vaccine> getVaccines() {
        return vaccines;
    }

    /**
     * @param vaccines the vaccines to set
     */
    public void setVaccines(ArrayList<Vaccine> vaccines) {
        this.vaccines = vaccines;
    }
    
    
}

File: hw/robalino/u2/hw18inheritance/animal.py
from datetime import date
from typing import List


class Animal:
    def __init__(self, id: int, name: str, born_on_date: date, weight: float, sex: str, vaccines: List['Vaccine']):
        self._id = id
        self._name = name
        self._born_on_date = born_on_date
        self._weight = weight
        self._sex = sex
        self._vaccines = vaccines

    def feed(self, food: 'Food'):
        print(f"Feeding the animal {self._name} with ---> {food}")

    def register(self):
        print(f"Registered animal {self._name}")

    def assign_cage(self, cage: 'Cage'):
        print(f"Assigning animal {self._name} to cage ---> {cage}")
        cage.get_animals().append(self)

    def provide_vaccination(self, vaccine: 'Vaccine'):
        print(f"Providing vaccine ---> {vaccine} <--- to {self._name}")

    @property
    def id(self) -> int:
        return self._id

    @id.setter
    def id(self, value: int):
        self._id = value

    @property
    def name(self) -> str:
        return self._name

    @name.setter
    def name(self, value: str):
        self._name = value

    @property
    def born_on_date(self) -> date:
        return self._born_on_date

    @born_on_date.setter
    def born_on_date(self, value: date):
        self._born_on_date = value

    @property
    def weight(self) -> float:
        return self._weight

    @weight.setter
    def weight(self, value: float):
        self._weight = value

    @property
    def sex(self) -> str:
        return self._sex

    @sex.setter
    def sex(self, value: str):
        self._sex = value

    @property
    def vaccines(self) -> List['Vaccine']:
        return self._vaccines

    @vaccines.setter
    def vaccines(self, value: List['Vaccine']):
        self._vaccines = value

    def __str__(self) -> str:
        return (f"Animal(id={self._id}, name={self._name}, born_on_date={self._born_on_date}, "
                f"weight={self._weight}, sex={self._sex}, vaccines={self._vaccines})")
